OptimizelySDK/Event/UserEventFactory.cs

```diff
@@ -76,15 +76,15 @@ public class UserEventFactory
             .WithProjectId(projectConfig.ProjectId)
             .WithAccountId(projectConfig.AccountId)
             .WithAnonymizeIP(projectConfig.AnonymizeIP)
-            .WithRevision(projectConfig.Revision)                
+            .WithRevision(projectConfig.Revision)
             .Build();
 
-            var variationKey = ""; 
-            var ruleKey = "";   
+            var variationKey = "";
+            var ruleKey = "";
             if (variation != null)
             {
                 variationKey = variation.Key;
-                ruleKey = activatedExperiment.Key;
+                ruleKey = activatedExperiment != null ? activatedExperiment.Key : "";
```

Comment:
activatedExperiment is also used in the ImpressionEvent builder, so we may also need to check null there.

Reply:
activatedExperiment?.Key ?? string.Empty